client/verta/verta/endpoint/resources.py

```diff
@@ -73,12 +78,23 @@ def _validate_memory(self, memory):
         if not re.match(r"^[0-9]+[e]?[0-9]*[E|P|T|G|M|K]?[i]?$", memory):
             raise ValueError(self.MEMORY_ERR_MSG)
 
+    def _validate_nvidia_gpu(self, nvidia_gpu):
+        if isinstance(nvidia_gpu, dict):
+            nvidia_gpu = NvidiaGPU._from_dict(nvidia_gpu)
```

Comment:
I don't think we should accept a `dict` here 🤔 

I'd recommend putting the dict conversion in `_from_dict()`, probably